packages/pyright-internal/src/analyzer/checker.ts

```diff
@@ -1534,6 +1534,18 @@ export class Checker extends ParseTreeWalker {
             }
         }
 
+        if (
+            reportAsUnused &&
+            this._fileInfo.ipythonMode === IPythonMode.CellDocs &&
+            node.parent?.nodeType === ParseNodeType.StatementList &&
+            node.parent.parent?.nodeType === ParseNodeType.Module &&
+            node.parent.statements[node.parent.statements.length - 1] === node
+        ) {
+            // Exclude an expression at the end of a notebook cell, as that is treated as
+            // the cell's value.
+            reportAsUnused = false;
```

Comment:
Does this work for sub expressions too? Like

```
df['MyColumn']
```

Reply:
It seems that this isn't considered an unused expression, so my change has no impact on it. I'm not sure why it isn't considered unused though.

Reply:
`_reportUnusedExpression` only creates diagnostics for what it considers to be `simpleExpressionTypes` and non-comprehension collections. Index expressions are not currently considered "simple". However, if they were, this change would work for your example.